.github/workflows/Tests.yaml

```diff
@@ -499,6 +499,31 @@ jobs:
             # Test `install` target with clang in Release mode because it uses
             # little disk space
             install: ON
+          # Test compiling with clang and CUDA
+          - compiler: clang-18
+            CUDA: ON
+            KOKKOS: ON
+            # Compile for NVIDIA Ampere architecture. Can't run this on
+            # GitHub-hosted runners because they don't have the GPU hardware,
+            # but we can test the compilation.
+            KOKKOS_ARCH: "AMPERE80"
+            # Compile in Release mode to speed up the build
+            build_type: Release
+            # Disable PCH as clang+cuda seems to have a problem with that
+            USE_PCH: OFF
+            # Disable ccache as it isn't forwarded to the clang+cuda correctly
+            USE_CCACHE: OFF
+            # Build with static libs because Kokkos CUDA doesn't support shared
+            # libs
+            BUILD_SHARED_LIBS: OFF
+            # Build only a subset of the code because compiling the full code is
+            # too slow and exceeds the available memory. We should extend this
+            # to the full code once we have a better solution for the memory
+            # issue. See 'Test nvcc support' step below.
+            unit_tests: OFF
```

Comment:
How true is this for clang+cuda? I expect that to be a lot better than NVCC, maybe so much so that we can actually build the unit tests?